lib/mix/lib/mix/dep/converger.ex

```diff
@@ -299,7 +299,7 @@ defmodule Mix.Dep.Converger do
         end
 
         cond do
-          in_upper? && other_opts[:override] ->
+          in_upper? && (other_opts[:override] || override_for?(other_opts, app)) ->
```

Comment:
I was most confused in this section of the converge logic.

If a dep passes this check once, does it never fail/process again? I initially had a test where I wanted dependency resolution to fail if `override_for: [:foo]` is specified, but `:baz` also required the override.